CMakeLists.txt

```diff
@@ -31,3 +31,32 @@ else()
   add_custom_target(check-format ${CLANG_FORMAT_NOT_FOUND_COMMAND_ARGS})
   add_custom_target(fix-format ${CLANG_FORMAT_NOT_FOUND_COMMAND_ARGS})
 endif()
+
+include(FindGit)
+find_program(CMAKE_FORMAT_PROGRAM cmake-format)
+
+if (Git_FOUND AND CMAKE_FORMAT_PROGRAM)
+  function (add_cmake_format_target name)
+    add_custom_target(${name}
+      COMMAND ${CMAKE_COMMAND} -DGIT_EXECUTABLE=${GIT_EXECUTABLE} -DCMAKE_FORMAT_PROGRAM=${CMAKE_FORMAT_PROGRAM}
+        -DCMAKE_FORMAT_TARGET=${name} -DBINARY_DIR="${CMAKE_CURRENT_BINARY_DIR}"
+        -P ${CMAKE_CURRENT_LIST_DIR}/cmake-format.cmake
+      WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+    )
+  endfunction()
+
+  add_cmake_format_target(cmake-format)
+  add_cmake_format_target(check-cmake-format)
+  add_cmake_format_target(fix-cmake-format)
```

Comment:
Good idea. One difference I forgot to mention, that I gather from the discussion at https://github.com/cheshirekow/cmake_format/issues/201, is that only committed C++ files are considered, whereas for CMake non-ignored untracked files are considered. Considering the merging of the tools under a single target, it'd make sense to only consider committed CMake files. What do you think?

Reply:
Yeah I agree, only tracked files should be considered as they could be restored in case the format was accidental or has unwanted consequences. (files with active changes are still considered though)